libsolidity/ast/Types.cpp

```diff
@@ -2924,7 +2924,10 @@ vector<tuple<string, TypePointer>> FunctionType::makeStackItems() const
 	{
 	case Kind::External:
 	case Kind::DelegateCall:
-		slots = {make_tuple("address", TypeProvider::address()), make_tuple("functionIdentifier", TypeProvider::fixedBytes(4))};
+		slots = {
+			make_tuple("address", TypeProvider::address()),
+			make_tuple("functionIdentifier", TypeProvider::uint(32))
```

Comment:
As discussed with @mijovic already: I think this should always have been ``uint32``, but we have to make sure that this doesn't break any existing code (although I don't think so) and that it'll always behave as desired (although I think it will). I'll look into it and comment again.

Reply:
Yes, after looking through the code I think this is the way to go, since this e.g. corresponds to the alignment of the identifier as decoded by the abi decoder.